### PR TITLE
Fix a possible division by zero

### PR DESCRIPTION
**Pull request by mikeSimonson**

### CODE REVIEW

File: DataCollector/DoctrineDataCollector.php [hunk 1]
```diff
@@ -222,15 +222,26 @@ public function getGroupedQueries()
             });
             $groupedQueries[$connection] = $connectionGroupedQueries;
         }
+
         foreach ($groupedQueries as $connection => $queries) {
             foreach ($queries as $i => $query) {
-                $groupedQueries[$connection][$i]['executionPercent'] = $query['executionMS'] / $totalExecutionMS * 100;
+                $groupedQueries[$connection][$i]['executionPercent'] =
+                    $this->executionTimePercentage($query['executionMS'], $totalExecutionMS );
```
Comment (stof) [inline DataCollector/DoctrineDataCollector.php]: I would not wrap the line here

File: DataCollector/DoctrineDataCollector.php [hunk 2]
```diff
@@ -222,15 +222,26 @@ public function getGroupedQueries()
...
             }
         }
 
         return $groupedQueries;
     }
 
+    private function executionTimePercentage($executionTimeMS, $totalExecutionTimeMS)
+    {
+        if ($totalExecutionTimeMS === 0.0 || $totalExecutionTimeMS === 0) {
```
Comment (e-moe) [inline DataCollector/DoctrineDataCollector.php]: s it safe to compare float values like this? Maybe we can use very small epsilon value e.g. if ($totalExecutionTimeMS < $eps) where $eps = 1e-5 or something like this

File: DataCollector/DoctrineDataCollector.php [hunk 3]
```diff
@@ -222,15 +222,26 @@ public function getGroupedQueries()
...
+            return 0;
```
Comment (greg0ire) [inline DataCollector/DoctrineDataCollector.php]: Returning 100 would make more sense IMO

File: DataCollector/DoctrineDataCollector.php (hunk 1 above)
Comment (greg0ire) [inline DataCollector/DoctrineDataCollector.php]: @stof the first parens is after 120 chars, so I think this was the right choice. @mikeSimonson you should remove the space after `$totalExecutionMS`.

**Comment by kimhemsoe** [comment]
Is there any remaining issues which is blocking a merge?

File: DataCollector/DoctrineDataCollector.php (hunk 2 above)
Comment (Majkl578) [reply DataCollector/DoctrineDataCollector.php]: Yes, this comparison looks dangerous.

**Pull request merged**